src/core_write.cpp

```diff
@@ -189,13 +191,20 @@ void TxToUniv(const CTransaction& tx, const uint256& hashBlock, UniValue& entry,
     entry.pushKV("weight", GetTransactionWeight(tx));
     entry.pushKV("locktime", (int64_t)tx.nLockTime);
 
-    UniValue vin(UniValue::VARR);
+    UniValue vin{UniValue::VARR};
+
+    // If available, use Undo data to calculate the fee. Note that txundo == nullptr
+    // for coinbase transactions and for transactions in pruned blocks.
```

Comment:
I think it's slightly better to be a bit more precise here. There may be other reason (in the future) why there is no undo data.
```suggestion
    // for coinbase transactions and for transactions where undo data is unavailable.
```